ecosystem/typescript/sdk/src/providers/aptos_client.ts

```diff
@@ -308,6 +316,91 @@ export class AptosClient {
     return builder.build(payload.function, payload.type_arguments, payload.arguments);
   }
 
+  /**
+   * Generates a fee payer transaction that can be signed and submitted to chain
+   *
+   * @param sender sender account address
+   * @param payload the transaction payload
+   * @param fee_payer the fee payer account
+   * @param secondary_signer_accounts an optional array of the secondary signers accounts
+   * @returns a fee payer raw transaction that can be signed and submitted to chain
+   */
+  async generateFeePayerTransaction(
+    sender: MaybeHexString,
+    payload: Gen.EntryFunctionPayload,
+    fee_payer: AptosAccount,
+    secondary_signer_accounts: Array<AptosAccount> = [],
+    options?: Partial<Gen.SubmitTransactionRequest>,
+  ): Promise<TxnBuilderTypes.FeePayerRawTransaction> {
+    const rawTxn = await this.generateTransaction(sender, payload, options);
+
+    const signers: Array<AccountAddress> = secondary_signer_accounts.map((account) =>
+      AccountAddress.fromHex(account.address()),
+    );
+
+    const feePayerTxn = new TxnBuilderTypes.FeePayerRawTransaction(
+      rawTxn,
+      signers,
+      AccountAddress.fromHex(fee_payer.address()),
+    );
+    return feePayerTxn;
+  }
+
+  /**
+   * Submits fee payer transaction to chain
+   *
+   * @param feePayerTransaction the fee payer transaction, of type FeePayerRawTransaction
```

Comment:
```suggestion
   * @param feePayerTransaction the raw transaction to be submitted, of type FeePayerRawTransaction
```

only to make it a little more clear. Up to you though, not sure if it's clearer this way or the other way